### PR TITLE
Fix library packages overlapping with framework packages

### LINKED ISSUE #23656: Using latest System.Memory package with netcoreapp2.0 fails due to S.N.V depenedency
If we bump up the version of the packages we reference in corefxlab to: 4.5.0-preview2-25628-01
https://github.com/dotnet/corefxlab/blob/master/tools/dependencies.props#L6-L8

..\.nuget\packages\microsoft.netcore.app\2.0.0\ref\netcoreapp2.0\System.Numerics.Vectors.dll
has version ver 4:1:3:0

But the latest System.Memory depends on 4.1.4
.assembly extern System.Numerics.Vectors
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:1:4:0
}

Our tests (which target netcoreapp2.0) fail with the following error:
Failed   System.Buffers.Tests.EqualityTests.SpansReferencingSameMemoryAreEqualInEveryAspect(bytes: [0, 0], start: 0, length: 2)
Error Message:
 System.IO.FileNotFoundException : Could not load file or assembly 'System.Numerics.Vectors, Version=4.1.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Stack Trace:
   at System.SpanHelpers.SequenceEqual(Byte& first, Byte& second, Int32 length)
   at System.Buffers.Tests.EqualityTests.SpansReferencingSameMemoryAreEqualInEveryAspect(Span`1 span, Span`1 pointingToSameMemory) in D:\GitHub\Fork\corefxlab\tests\System.Buffers.Primitives.Tests\EqualityTests.cs:line 163
   at System.Buffers.Tests.EqualityTests.SpansReferencingSameMemoryAreEqualInEveryAspect(Byte[] bytes, Int32 start, Int32 length) in D:\GitHub\Fork\corefxlab\tests\System.Buffers.Primitives.Tests\EqualityTests.cs:line 45

cc @weshaggard, @ericstj, @KrzysztofCwalina, @shiftylogic, @mellinoe

### PR DESCRIPTION
**Pull request by ericstj**
Many library packages overlap with assemblies we include in our framework packages.

In the first release we handled this by preventing the library package from applying any
assets.

This becomes problematic if the reference assembly needs to version, and especially
if the library wants to add API.  The new version will be greater than what was previously
shipped inbox, and we wouldn't want to restrict the new version just to future TFMs.

To address this problem I've made a couple fixes:
1. For packages which are shipping brand new API that's not present in desktop, we'll
omit any placeholders and allow conflict resolution to decide if the library package
or framework package should be used (based on version).  In some cases this required
bringing some configurations into the package so that it could always be "up to date".

2. For packages which are just exposing API in desktop, and thus already tied/anchored
to netstandard versions, we'll freeze the reference assembly version for the non-desktop
builds.  This ensures that the reference assembly will never move past the version that's
inbox in the framework package.

Update BuildToolsVersion to 2.0.0-prerelease-01931-01

Fixes #23656

### CODE REVIEW

**Comment by ericstj** [comment]
test UWP NETNative x86 Release Build

**Comment by ericstj** [comment]
Note that any of these cases where I froze the reference to a version that's less than current we may want to port to UWP release branch (though, it would just mean that UWP's own copy would always win over library package)

File: src/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.csproj [hunk 1]
```diff
@@ -3,7 +3,10 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{C4D6F1F4-DC7E-4756-9A88-171A8B1F1E26}</ProjectGuid>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netfx'">true</IsPartialFacadeAssembly>
+    <!-- Must match version supported by frameworks which support 4.2.* inbox.
+         Can be removed when API is added and this assembly is versioned to 4.1.* -->
```
Comment (weshaggard) [inline src/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.csproj]: versioned to 4.4.*?

**Comment by weshaggard** [comment]
While I think I understand how this currently helps I don't really like it as it doesn't seem maintainable. How do we make sure that we stay safe here? 

I'm fine with you moving forward with this approach but I do want us to sit down and talk through this problem more to explore all options.

**Comment by ericstj** [comment]
Windows failed with [helix download issue](https://mc.dot.net/#/user/ericstj/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/75962dae7f338445b57d7d276d018774e1dd2e46/workItem/System.Xml.XmlDocument.Tests/wilogs) /cc @Chrisboh 
```
2017-09-01 16:40:19,052: INFO: executor(331): _get_unzipped_file_paths: Reading zip file:\\?\C:\dotnetbuild\work\298a4d42-f67b-4167-b0f7-8f0dd25510ac\Work\6750c854-8cbe-41f0-97e1-ec5be12a4c41\System.Xml.XmlDocument.Tests.zip
2017-09-01 16:40:19,052: ERROR: executor(92): run: Unhandled exception attempting to download payloads
```

**Comment by ericstj** [comment]
Test Windows x64 Debug Build

**Comment by maririos** [comment]
> Windows failed with helix download issue /cc @Chrisboh 

I'll take a look at it. Issue https://github.com/dotnet/core-eng/issues/1696

**Pull request merged**